lib/perl/Genome/Config/Profile/Item.pm

```diff
@@ -155,7 +155,8 @@ sub _copy_file_to_allocation {
         $allocation->reallocate();
     }
     else {
-        my $config_file = 'model.' . $self->id . $filename . '.yaml';
+        my ($file_name, $file_dirs, $file_suffix) = File::Basename::fileparse($config_path, qr/\.[^.]+/);
+        my $config_file = 'model.' . $self->id . '.' . $file_name . '.yaml';
```

Comment:
This might make it easier to avoid missing `.`s if the name changes again later.
```suggestion
        my $config_file = join('.', 'model', $self->id, $file_name, 'yaml');
```